docs: add drone ci to list of supported services (#177)
localstack/docs: content/en/ci/drone-ci/index.md

---
title: "Drone CI"
tags: ["continuous-integration", "ci", "continuous-delivery", "testing"] 
weight: 5
description: >
  Use LocalStack in [Drone CI](https://drone.io/)
---

This guide shows you how to start LocalStack in a Drone CI pipeline.

## Drone CI Pipelines

There are a few restrictions in Drone CI Pipelines that make it hard to customize the behavior of Docker. For example, mounting the host machine Docker socket is considered highly insecure, and hence alleviated privileges are required to run commands like `localstack wait`. Learn more about configuring Docker for Drone CI pipelines over [their official documentation](https://docs.drone.io/pipeline/docker/overview/).

> A possible work-around to mount the Docker socket (`/var/run/docker.sock`) into the container to communicate with the docker daemon on the host is by exposing the port `2375`, which is reachable in the main LocalStack container. However, it requires changes to the typical Docker-Compose setup.

Reviewer: ```suggestion
A possible alternative to mounting the Docker socket (`/var/run/docker.sock`) into the container to communicate with the Docker daemon on the host is to expose port `2375` and make it reachable in the main LocalStack container. However, it requires changes to the typical Docker-Compose setup.
```

Reviewer: Not sure if my suggestion matches what we're intending to express here - feel free to adjust as you see fit! 👍 

Could we perhaps add a small config snippet that illustrates how this can be achieved, to make the Docker daemon available? 

Otherwise, if we cannot easily get the LocalStack container to talk to the Docker engine (and spin up new containers), it would be great if we could document the limitations (e.g., that only the `local` Lambda executor will work in that case, i.e., users should configure `LAMBDA_EXECUTOR=local`).
